GitCommands/Settings/Setting.cs

```diff
@@ -4,91 +4,131 @@
 
 namespace GitCommands.Settings
 {
-    public abstract class Setting<T>
+    public interface ISetting<T>
     {
-        public readonly SettingsPath SettingsSource;
-        public readonly T DefaultValue;
-        public readonly string Name;
+        event EventHandler Updated;
 
-        public event EventHandler Updated;
+        SettingsPath SettingsSource { get; }
 
-        protected Setting(string name, SettingsPath settingsSource, T defaultValue)
+        string Name { get; }
+
+        T DefaultValue { get; }
+
+        T ValueOrDefault { get; }
+
+        T Value { get; set; }
+
+        string FullPath { get; }
+
+        bool ValueIsEmpty(T value);
+    }
+
+    public static class Setting
+    {
+        public static ISetting<string> Create(SettingsPath settingsSource, string name, string defaultValue)
```

Comment:
Would `Bind` be more descriptive of a nature of the operation here? That is, we _create a binding to an existing setting_ vs we _create a new setting_.

Reply:
In this case, we are creating. We bind to the UI elsewhere. Just a factory method.